### PR TITLE
gossip: Use the internal utxoset to verify channels and store capacity in routing table

### PR DESCRIPTION
**Pull request by cdecker**
This PR shortcircuits the txout check to use the internal utxoset if present, saving us from having to call out and parse a block. In addition we now annotate the channels in our routing view to have the total capacity of the channel, hopefully allowing us to make more informed routing decisions in the future.

### CODE REVIEW

**Comment by ZmnSCPxj** [comment]
ACK 38b95c70d4042cb746e359a7528c53de127853d2

File: lightningd/gossip_control.c [hunk 1]
```diff
@@ -66,15 +66,19 @@ static void got_txout(struct bitcoind *bitcoind,
...
 
 	/* output will be NULL if it wasn't found */
-	if (output)
+	if (output) {
 		script = output->script;
-	else
+		satoshis = output->amount;
+	} else {
 		script = NULL;
+	}
```
Comment (rustyrussell) [inline lightningd/gossip_control.c]: I have a minor preference for setting things to NULL in an if branch, over initialization.  That way the compiler will (sometimes) warn you when it's used unset, if anything is missed.  See how script is done here, for example.

File: lightningd/gossip_control.c [hunk 2]
```diff
@@ -440,6 +440,7 @@ static void json_listchannels_reply(struct subd *gossip UNUSED, const u8 *reply,
 		json_add_num(response, "flags", entries[i].flags);
 		json_add_bool(response, "active", entries[i].active);
 		json_add_bool(response, "public", entries[i].public);
+		json_add_num(response, "satoshis", entries[i].satoshis);
```
Comment (rustyrussell) [inline lightningd/gossip_control.c]: json_add_u64...

**Comment by cdecker** [comment]
Addressed the feedback from @rustyrussell, ready for round 2 :wink:

**Pull request merged**